hyp3_testing/opera_compare.py

```diff
@@ -0,0 +1,325 @@
+        list_dataset_1, list_attrs_1 = get_list_dataset_attrs_keys(hdf5_in_1)
+        set_dataset_1 = set(list_dataset_1)
+        set_attrs_1 = set(list_attrs_1)
+
+        list_dataset_2, list_attrs_2 = get_list_dataset_attrs_keys(hdf5_in_2)
+        set_dataset_2 = set(list_dataset_2)
+        set_attrs_2 = set(list_attrs_2)
+
+        intersection_set_dataset = set_dataset_1.intersection(set_dataset_2)
+        assert len(intersection_set_dataset) == len(set_dataset_1)
+        assert len(intersection_set_dataset) == len(set_dataset_2)
+
+        intersection_set_attrs = set_attrs_1.intersection(set_attrs_2)
+        assert len(intersection_set_attrs) == len(set_attrs_1)
+        assert len(intersection_set_attrs) == len(set_attrs_2)
+
+        # Proceed with checking the values in dataset,
+        # regardless of the agreement of their structure.
+        for key_dataset in intersection_set_dataset:
+            compare_hdf5_elements(hdf5_in_1, hdf5_in_2, key_dataset, is_attr=False)
+
+        # Proceed with checking the values in attributes,
+        # regardless of the agreement of their structure.
+        for key_attr in intersection_set_attrs:
+            compare_hdf5_elements(hdf5_in_1, hdf5_in_2, key_attr, is_attr=True)
+
+
+def elements_equal(root, e1, e2):
+    nested = ['{http://earthdata.nasa.gov/schema/eos}value']
+    assert e1.tag == e2.tag, f'Tag mismatch at {e1.tag}: {e1.tag} != {e2.tag}'
+    assert len(e1) == len(e2), f'Children count mismatch at {e1.tag}: {len(e1)} != {len(e2)}'
+    assert e1.attrib == e2.attrib, f'Attribute mismatch at {e1.tag}: {e1.attrib} != {e2.attrib}'
+    assert e1.nsmap == e2.nsmap, f'Namespace mismatch at {e1.tag}: {e1.nsmap} != {e2.nsmap}'
+
+    full_path = root.getroottree().getpath(e1)
+    if full_path not in LIST_EXCLUDE_COMPARISON_XML:
+        # if not (e1.text or '').strip() == (e2.text or '').strip():
+        assert (e1.text or '').strip() == (e2.text or '').strip(), (
+            f'Text mismatch at {full_path}: {e1.text} != {e2.text}'
+        )
+
+    # Recursively compare children
+    for c1, c2 in zip(e1, e2):
+        elements_equal(root, c1, c2)
+
+
+def compare_rtc_iso_xmls(file1, file2):
+    root1 = etree.parse(file1).getroot()
+    root2 = etree.parse(file2).getroot()
+    elements_equal(root1, root1, root2)
+
+
+def _compare_rtc_s1_metadata(metadata_1, metadata_2):
+    set_1_m_2 = set(metadata_1.keys()) - set(metadata_2.keys())
+    assert set_1_m_2 == set()
+    set_2_m_1 = set(metadata_2.keys()) - set(metadata_1.keys())
+    assert set_2_m_1 == set()
+    for k1, v1 in metadata_1.items():
+        if k1 in LIST_EXCLUDE_COMPARISON_IMAGE:
+            continue
+        v2 = metadata_2[k1]
+        assert v2 == v1, f'Values for key {k1} do not match ({v1} | {v2})'
+
+
+def compare_rtc_s1_products(file_1: Path, file_2: Path):
+    assert file_1.exists()
+    assert file_2.exists()
+
+    # TODO: compare projections ds.GetProjection()
```

Comment:
Is this `TODO` a blocker for this PR?

Reply:
No, this is actually a TODO the OPERA team left. I'll remove it.